arbnode/message_pruner.go

```diff
@@ -88,100 +89,42 @@ func (m *MessagePruner) prune(ctx context.Context) time.Duration {
+	var allPrunedKeys [][]byte
+	prunedKeys, err := deleteFromLastPrunedUptoEndKey(inboxTrackerDb, sequencerBatchMetaPrefix, endBatchCount)
 	if err != nil {
-		log.Warn("error checking last pruned batch metadata: %w", err)
-	} else if hasKey {
-		data, err := inboxTrackerDb.Get(lastPrunedSequencerBatchMetaKey)
-		if err != nil {
-			log.Warn("error fetching last pruned batch metadata: %w", err)
-		} else {
-			err = rlp.DecodeBytes(data, &startBatchCount)
-			if err != nil {
-				log.Warn("error decoding last pruned batch metadata: %w", err)
-			}
-		}
+		log.Error("error deleting batch metadata: %w", err)
+		return
 	}
-	if endBatchCount > startBatchCount {
-		err := deleteFromRange(inboxTrackerDb, sequencerBatchMetaPrefix, startBatchCount, endBatchCount-1)
-		if err != nil {
-			log.Error("error deleting batch metadata: %w", err)
-			return
-		}
-		endBatchCountValue, err := rlp.EncodeToBytes(endBatchCount - 1)
-		if err != nil {
-			log.Error("error encoding end batch count: %w", err)
-			return
-		}
-		err = inboxTrackerDb.Put(lastPrunedSequencerBatchMetaKey, endBatchCountValue)
-		if err != nil {
-			log.Error("error storing last pruned batch metadata: %w", err)
-			return
-		}
+	allPrunedKeys = append(allPrunedKeys, prunedKeys...)
+
+	prunedKeys, err = deleteFromLastPrunedUptoEndKey(transactionStreamerDb, messagePrefix, uint64(endBatchMetadata.MessageCount))
+	if err != nil {
+		log.Error("error deleting batch metadata: %w", err)
+		return
 	}
+	allPrunedKeys = append(allPrunedKeys, prunedKeys...)
 
-	startMessageCount := uint64(1)
-	hasKey, err = transactionStreamerDb.Has(lastPrunedMessageKey)
+	prunedKeys, err = deleteFromLastPrunedUptoEndKey(inboxTrackerDb, rlpDelayedMessagePrefix, endBatchMetadata.DelayedMessageCount)
 	if err != nil {
-		log.Warn("error checking last pruned message: %w", err)
-	} else if hasKey {
-		data, err := transactionStreamerDb.Get(lastPrunedMessageKey)
-		if err != nil {
-			log.Warn("error fetching last pruned message: %w", err)
-		} else {
-			err = rlp.DecodeBytes(data, &startMessageCount)
-			if err != nil {
-				log.Warn("error decoding last pruned message: %w", err)
-			}
-		}
+		log.Error("error deleting batch metadata: %w", err)
+		return
 	}
-	if uint64(endBatchMetadata.MessageCount) > startMessageCount {
-		err := deleteFromRange(transactionStreamerDb, messagePrefix, startMessageCount, uint64(endBatchMetadata.MessageCount)-1)
-		if err != nil {
-			log.Error("error deleting last batch messages: %w", err)
-		}
-		endMessageCountValue, err := rlp.EncodeToBytes(uint64(endBatchMetadata.MessageCount) - 1)
-		if err != nil {
-			log.Error("error encoding end message count: %w", err)
-			return
-		}
-		err = transactionStreamerDb.Put(lastPrunedMessageKey, endMessageCountValue)
-		if err != nil {
-			log.Error("error storing last pruned message: %w", err)
-			return
-		}
+	allPrunedKeys = append(allPrunedKeys, prunedKeys...)
+
+	if len(allPrunedKeys) > 0 {
+		log.Info("Pruned keys:", allPrunedKeys)
```

Comment:
This is going to be an impossibly long log line. I'm not sure even storing all the keys in memory is viable. I'd instead print the first and last indices pruned. The `log.Info` format also accepts parameters like `log.Info("base message", "param1", param1Value, "param2", param2Value)`. It complains and doesn't format the log message right if you don't supply the key for the value you print out.

Reply:
done.